soc/xtensa/intel_adsp/common/include/cavs_hda.h

```diff
@@ -0,0 +1,298 @@
+/* Copyright (c) 2022 Intel Corporation
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_CAVS_HDA_H
+#define ZEPHYR_INCLUDE_CAVS_HDA_H
+
+#include <arch/xtensa/cache.h>
+#include <kernel.h>
+#include <device.h>
+#include <cavs-shim.h>
+#include <cavs-mem.h>
+
+/**
+ * @brief HDA stream functionality for cAVS
+ *
+ * Provides low level calls to support cAVS HDA streams with
+ * minimal abstraction that allows testing the hardware
+ * and its demands separately from the intended DMA API
+ * usage. The only requirement is that you define the base
+ * addresses, the stream count, and the size of the ip blocks.
+ */
+
+#ifndef HDA_HOST_OUT_BASE
+#error Must define HDA_HOST_OUT_BASE
+#endif
+#ifndef HDA_HOST_IN_BASE
+#error Must define HDA_HOST_IN_BASE
+#endif
+#ifndef HDA_STREAM_COUNT
+#error Must define HDA_STREAM_COUNT
+#endif
+#ifndef HDA_REGBLOCK_SIZE
+#error Must define HDA_REGBLOCK_SIZE
+#endif
+
+/* The read/write positions are masked to 24 bits */
+#define HDA_RWP_MASK 0x00FFFFFF
+
+/* Buffers must be 128 byte aligned, this mask enforces that */
+#define HDA_ALIGN_MASK 0xFFFFFF80
+
+
+#define HDA_ADDR(base, stream) (base + stream*HDA_REGBLOCK_SIZE)
```

Comment:
`((base) + (stream)*HDA_REGBLOCK_SIZE)` ? It helps avoid wrong behavior if someone calls it `HDA_ADDR(base, stream + offset)`